armi/reactor/components/__init__.py

```diff
@@ -464,24 +472,40 @@ def getVolume(self):
         vol = UnshapedComponent.getVolume(self)
         return vol
 
-    def getComponentArea(self, cold=False):
+    def getComponentArea(self, cold=False, Tc=None):
         """
         Get the area of this component in cm^2.
 
         Parameters
         ----------
         cold : bool, optional
             If True, compute the area with as-input dimensions, instead of thermally-expanded.
+        Tc : float, optional
+            Temperature in C to compute the area at
         """
+        if cold and Tc is not None:
+            raise ValueError(
+                f"Cannot compute component area at {Tc} and cold dimensions simultaneously."
+            )
+
         if cold:
             # At cold temp, the DerivedShape has the area of the parent minus the other siblings
-            parentArea = self.parent.getArea()
+            parentArea = self.parent.getMaxArea()
             # NOTE: the assumption is there is only one DerivedShape in each Component
             siblings = sum(
                 [c.getArea(cold=True) for c in self.parent if type(c) != DerivedShape]
             )
             return parentArea - siblings
 
+        if Tc is not None:
+            # The DerivedShape has the area of the parent minus the other siblings
+            parentArea = self.parent.getMaxArea()
+            # NOTE: the assumption is there is only one DerivedShape in each Component
```

Comment:
```suggestion
            # NOTE: Here we assume there is one-and-only-one DerivedShape in each Component
```

Reply:
updated here and the next comment which was the same.